pandas/core/frame.py

```diff
@@ -5640,7 +5637,9 @@ def count(self, axis=0, level=None, numeric_only=False):
         if len(frame._get_axis(axis)) == 0:
             result = Series(0, index=frame._get_agg_axis(axis))
         else:
-            if frame._is_mixed_type:
```

Comment:
this whole count things should be cleaned up anyhow, can you add an issue about this, this optimization is not worth it an makes code more complex (not about your changes, but the mixed type stuff)